### ISSUE TITLE
NA related error in `plot_cast_point(with_census = TRUE)`?

### ISSUE THREAD

**Issue by ethanwhite** [weecology/portalcasting#121]
Following the Getting Started vignette I run into the following error:

```R
> plot_cast_point(with_census = TRUE)
Error in read_cast(tree, cast_type = cast_type, cast_date = cast_date) : 
  forecasts from NA not available
In addition: Warning message:
In max.default(numeric(0), na.rm = FALSE) :
  no non-missing arguments to max; returning -Inf
```

I'm guessing this was something to do with NA being interpreted as a null value given the error, but it also triggers when setting `species = "DM"`.

**Comment by ethanwhite** [comment]
This is using the current `master` of `portalr` based on the `devtools` install given at the top of the vignette.

**Comment by ethanwhite** [comment]
This error is particularly weird because it supposedly occurs in `read_cast`, but the only use of in `plot_cast_point` occurs before `with_census` is used https://github.com/weecology/portalcasting/blob/master/R/figures.R#L426

**Comment by juniperlsimonis** [comment]
ohh! this is because the default setting is to not download the whole set of predictions from the repo, so there aren't any predictions that have subsequent observations in your local version of it. 

this brings up two components of the issue:
1. the downloading of the past predictions. i had a very difficult time getting that to work reliably cleanly. i could definitely use some help figuring out how to make the downloading of the existing predictions from the repo work better. (more reliably and with more specific control, rather than just "all or nothing")
2. the vignette/getting started space. i don't think we should force on a new user by default downloading the whole set of past predictions, as that's just a lot to process. we could set it up so that some subset get downloaded, but that is currently not available. in the meantime, however, it is totally possible to set up a local forecast that would let you make the plot you couldn't make, by taking advantage of hindcasting...basically going back in time to forecast data that have been collected already. 

what do you think about how to approach this?

**Comment by ethanwhite** [comment]
> the downloading of the past predictions. i had a very difficult time getting that to work reliably cleanly. i could definitely use some help figuring out how to make the downloading of the existing predictions from the repo work better. (more reliably and with more specific control, rather than just "all or nothing")

That does sound tricky. Can you open an issue for this one with a description of what you've already encountered to give us a place to discuss it?

**Comment by ethanwhite** [comment]
> the vignette/getting started space. i don't think we should force on a new user by default downloading the whole set of past predictions, as that's just a lot to process. we could set it up so that some subset get downloaded, but that is currently not available. in the meantime, however, it is totally possible to set up a local forecast that would let you make the plot you couldn't make, by taking advantage of hindcasting...basically going back in time to forecast data that have been collected already.

Hmm, yeah, this is tricky. I guess, off the top of my head, I'd lean towards just not including that spot in the Getting Started document. Since we don't see the new user accessing this kind of data then that might be a good indication that we want this is a separate vignette. I don't feel strongly about this though.

**Comment by juniperlsimonis** [comment]
downloading ease, and thus downstream plotting addressed with #129
